src/NHSD.BuyingCatalogue.Organisations.Api/Services/EmailAddress.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System;
+using MimeKit;
+
+namespace NHSD.BuyingCatalogue.Organisations.Api.Services
+{
+    /// <summary>
+    /// An e-mail address.
+    /// </summary>
+    internal sealed class EmailAddress
+    {
+        private string address;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="EmailAddress"/> class.
+        /// </summary>
+        /// <remarks>Required by <see cref="System.Text.Json"/>.</remarks>
+        public EmailAddress()
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="EmailAddress"/> class
+        /// with the given name and address.
+        /// </summary>
+        /// <param name="name">An optional </param>
+        /// <param name="address"></param>
+        internal EmailAddress(string name, string address)
+        {
+            Name = name;
+            Address = address;
+        }
+
+        /// <summary>
+        /// Gets or sets the display name of the address.
+        /// </summary>
+        /// <remarks>An optional display name, for example
+        /// Buying Catalogue Team.</remarks>
+        public string Name { get; set; }
+
+        /// <summary>
+        /// Gets or sets the actual e-mail address.
+        /// </summary>
+        /// <exception cref="ArgumentNullException"><paramref name="value"/> is <see langref="null"/>.</exception>
+        /// <exception cref="ArgumentException"><paramref name="value"/> is empty or white space.</exception>
+        public string Address
+        {
+            get => address;
+            set
+            {
+                if (value == null)
```

Comment:
Any reason for the two statements here? `String.IsNullOrEmpty()` will carry out the first conditional check.

Reply:
There are two separate statements so that the most specific exception type can be thrown: `ArgumentNullException` in the null case and `ArgumentException` in the other cases.

While `string.IsNullOrWhiteSpace` and `string.IsNullOrEmpty` obviously both check for a null string we'd end up throwing `ArgumentException` in every case. While it's a small difference, I think it's preferable to throw the most explicit exception we can – it just means we end up covering the null case twice. It would be good if MS could provide some nice solution for this but I'm not aware of one.